modules/imgproc/src/drawing.cpp

```diff
@@ -1579,7 +1579,7 @@ Circle( Mat& img, Point center, int radius, const void* color, int fill )
                     ICV_HLINE( tptr, x11, x12, color, pix_size );
             }
 
-            if( (unsigned)y12 < (unsigned)size.height )
+            if( y12 >= 0 && y12 < size.height )
```

Comment:
Though standard C++ promotion happens, y12 and size.height have different types so the compiler might complain. Casting to (int64_t) will it should be fine with our CI.